code/datums/martial/sleeping_carp.dm

```diff
@@ -209,10 +207,14 @@
 	righthand_file = 'icons/mob/inhands/weapons/staves_righthand.dmi'
 	block_chance = 50
 
-/obj/item/twohanded/bostaff/update_icon_state()
-	icon_state = "bostaff[wielded]"
+/obj/item/staff/bostaff/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/two_handed, force_unwielded=10, force_wielded=24)
+
+/obj/item/staff/bostaff/update_icon_state()
+	icon_state = "bostaff[SEND_SIGNAL(src, COMSIG_IS_TWOHANDED_WIELDED)]"
```

Comment:
Pass in the icon_state that should be used while help two-handed in the component and listen for the update_icon signal

Reply:
Now managed by component